traffic_ops/testing/api/v4/cdn_dnsseckeys_test.go

```diff
@@ -0,0 +1,156 @@
+
+	res, _, err = TOSession.GetCDNDNSSECKeys(firstCDN.Name, client.RequestOptions{})
+	assert.RequireNoError(t, err, "Unexpected error getting CDN DNSSEC keys: %v - alerts: %+v", err, res.Alerts)
+
+	newKeys := res.Response
+
+	if reflect.DeepEqual(originalKeys, newKeys) {
+		t.Errorf("Generating CDN DNSSEC keys - expected: original keys to differ from new keys, actual: they are the same")
+	}
+
+	kskReq := tc.CDNGenerateKSKReq{
+		ExpirationDays: util.Uint64Ptr(30),
+	}
+	originalKSK := newKeys
+	resp, _, err = TOSession.GenerateCDNDNSSECKSK(firstCDN.Name, kskReq, client.RequestOptions{})
+	assert.NoError(t, err, "Unexpected error generating DNSSEC KSK: %v - alerts: %+v", err, resp.Alerts)
+
+	res, _, err = TOSession.GetCDNDNSSECKeys(firstCDN.Name, client.RequestOptions{})
+	assert.NoError(t, err, "Unexpected error getting CDN DNSSEC keys: %v - alerts: %+v", err, res.Alerts)
+
+	if _, ok := res.Response[firstCDN.Name]; !ok {
+		t.Fatalf("getting CDN DNSSEC keys - expected: key %s, actual: missing", firstCDN.Name)
+	}
+	newKSK := res.Response
+	if reflect.DeepEqual(originalKSK[firstCDN.Name].KSK, newKSK[firstCDN.Name].KSK) {
+		t.Error("Generating CDN DNSSEC KSK - Expected: KSK to be different, Actual: KSK is the same")
+	}
+	if !reflect.DeepEqual(originalKSK[firstCDN.Name].ZSK, newKSK[firstCDN.Name].ZSK) {
+		t.Error("Generating CDN DNSSEC KSK - Expected: ZSK to be equal, Actual: ZSK is different")
+	}
+
+	// ensure that when DNSSEC is enabled on a CDN, creating a new DS will generate DNSSEC keys for that DS:
+	if !cdn.DNSSECEnabled {
+		cdn.DNSSECEnabled = true
+		resp, _, err := TOSession.UpdateCDN(cdn.ID, cdn, client.RequestOptions{})
+		assert.NoError(t, err, "Unexpected error updating CDN: %v - alerts: %+v", err, resp.Alerts)
+
+		defer func() {
+			cdn.DNSSECEnabled = false
+			resp, _, err := TOSession.UpdateCDN(cdn.ID, cdn, client.RequestOptions{})
+			assert.NoError(t, err, "Unexpected error updating CDN: %v - alerts: %+v", err, resp.Alerts)
+		}()
+	}
+
+	opts.QueryParameters.Set("name", "HTTP")
+	types, _, err := TOSession.GetTypes(opts)
+	assert.RequireNoError(t, err, "Unexpected error getting Types filtered by name 'HTTP': %v - alerts: %+v", err, types.Alerts)
+	assert.RequireEqual(t, len(types.Response), 1, "Expected exactly one Type to exist with name 'HTTP', found: %d", len(types.Response))
+
+	dsXMLID := "testdnssecgen"
+	customDS := getCustomDS(cdn.ID, types.Response[0].ID, dsXMLID, "cdn", "https://testdnssecgen.example.com", dsXMLID)
+	ds, _, err := TOSession.CreateDeliveryService(customDS, client.RequestOptions{})
+	assert.NoError(t, err, "Unexpected error creating Delivery Service: %v - alerts: %+v", err, ds.Alerts)
+	assert.Equal(t, len(ds.Response), 1, "Expected creating a Delivery Service to create exactly one Delivery Service, Traffic Ops returned: %d", len(ds.Response))
+	assert.NotNil(t, ds.Response[0].ID, nil, "Traffic Ops returned a representation for a created Delivery Service with null or undefined ID")
```

Comment:
This should use `RequireEqual` and `RequireNotNil` on lines 142 and 143, respectively